packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -92,7 +92,9 @@ const BlockInspector = ( { showNoBlockSelectedMessage = true } ) => {
 			blockType: _blockType,
 			topLevelLockedBlock:
 				__unstableGetContentLockingParent( _selectedBlockClientId ) ||
-				( getTemplateLock( _selectedBlockClientId ) === 'contentOnly'
+				( getTemplateLock( _selectedBlockClientId ) === 'contentOnly' ||
+				( _selectedBlockName === 'core/block' &&
+					window.__experimentalPatternPartialSyncing )
```

Comment:
I think we should make sure we consider reviewing this and the similar code (in `use-in-between-inserter.js` and `selectors.js` code down the road, as there's a lot of hard coding of values to bend things into a working state. Feels like there should be a better way.

The gist of the problem seems to be that none of the block editing modes seem to be suitable for patterns.

Reply:
I'd love to understand the problem more here.

Reply:
At the moment the PR uses `useBlockEditingMode` to prevent editing inside the pattern block, but it doesn't support exactly what's needed for the pattern block. 'disabled' locks down too much. 'contentOnly' also isn't right. At the moment the PR sets the inner blocks of the pattern to 'disabled' or 'contentOnly' as needed, but that still results in things like the sibling inserter showing.

This part still needs some investigation.